extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/AddContainerPortDecorator.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.kubernetes.deployment;
+
+import io.dekorate.deps.kubernetes.api.model.ContainerBuilder;
+import io.dekorate.kubernetes.annotation.Protocol;
+import io.dekorate.kubernetes.config.Port;
+import io.dekorate.kubernetes.decorator.AddSidecarDecorator;
+import io.dekorate.kubernetes.decorator.ContainerDecorator;
+import io.dekorate.kubernetes.decorator.Decorator;
+import io.dekorate.kubernetes.decorator.ResourceProvidingDecorator;
+
+/**
+ * Adapted from Dekorate's {@link io.dekorate.kubernetes.decorator.AddPortDecorator} in order to
+ * address the limitation that the port was not being added (due to missing metadata)
+ */
+class AddContainerPortDecorator extends Decorator<ContainerBuilder> {
```

Comment:
It's not obvious to me what's wrong with the upstream version.

Also, keep in mind that this implementation will result in the port being applied to all containers of all deployments (including init containers, sidecars or any other kind of container generated, or found under /src/main/kubernetes).

Reply:
The upstream version doesn't apply the port because the object contains no metadata.

I am pretty certain that this decorator doesn't apply to all containers and if I am not mistaken, my test verifies that